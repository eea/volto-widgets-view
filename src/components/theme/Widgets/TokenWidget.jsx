import React from 'react';
import cx from 'classnames';
import { Tags } from '@plone/volto/components';

export const TokenWidget = ({ children, className }) => (
  <div className={cx(className, 'token', 'widget')}>
    <Tags tags={children} />
  </div>
);
