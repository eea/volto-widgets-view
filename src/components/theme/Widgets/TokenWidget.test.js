import React from 'react';
import renderer from 'react-test-renderer';
import { TokenWidget } from './TokenWidget';
import { MemoryRouter } from 'react-router-dom';

describe('TokenWidget', () => {
  it('renders an empty view widget based on @plone/volto Tags component', () => {
    const component = renderer.create(
      <MemoryRouter>
        <TokenWidget />
      </MemoryRouter>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders view widget based on @plone/volto Tags component', () => {
    const component = renderer.create(
      <MemoryRouter>
        <TokenWidget className="meta tags">{['foo', 'bar']}</TokenWidget>
      </MemoryRouter>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
