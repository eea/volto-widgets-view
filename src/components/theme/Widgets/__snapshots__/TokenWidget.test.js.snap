// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenWidget renders view widget based on @plone/volto Tags component 1`] = `
<div
  className="meta tags token widget"
>
  <div
    className="ui container tags"
  >
    <a
      className="ui label"
      href="/search?Subject=foo"
      onClick={[Function]}
    >
      foo
    </a>
    <a
      className="ui label"
      href="/search?Subject=bar"
      onClick={[Function]}
    >
      bar
    </a>
  </div>
</div>
`;
